pkg/alertmanager/operator.go

```diff
@@ -1790,3 +1797,28 @@ func ListOptions(name string) metav1.ListOptions {
 func tlsAssetsSecretName(name string) string {
 	return fmt.Sprintf("%s-tls-assets", prefixedName(name))
 }
+
+func GetAppliedConfigFromCondition(c *monitoringv1.Condition) *monitoringv1ac.ConditionApplyConfiguration {
+	return monitoringv1ac.Condition().
+		WithType(c.Type).
+		WithStatus(c.Status).
+		WithLastTransitionTime(c.LastTransitionTime).
+		WithReason(c.Reason).
+		WithMessage(c.Message).
+		WithObservedGeneration(c.ObservedGeneration)
+}
+
+func GetAppliedConfigStatusFromStatus(a *monitoringv1.Alertmanager, asac *monitoringv1ac.AlertmanagerStatusApplyConfiguration, values ...*monitoringv1ac.ConditionApplyConfiguration) {
```

Comment:
not sure if like this is best option because in GetAppliedConfigStatusFromStatus are monitoringv1ac and monitoringv1 values. Check the new changes please